.codegen/model.go.tmpl

```diff
@@ -35,20 +35,23 @@ type {{.PascalName}} struct {
 {{- end -}}
 
 {{- define "type" -}}
-	{{- if not . }}any /* ERROR */
-	{{- else if .IsExternal }}{{.Package.Name}}.{{.PascalName}}
-	{{- else if .IsAny}}any
-	{{- else if .IsEmpty}}{{.PascalName}}
-	{{- else if .IsString}}types.String
-	{{- else if .IsBool}}types.Bool
-	{{- else if .IsInt64}}types.Int64
-	{{- else if .IsFloat64}}types.Float64
-	{{- else if .IsInt}}types.Int64
-	{{- else if .IsByteStream}}io.ReadCloser
-	{{- else if .ArrayValue }}[]{{template "type" .ArrayValue}}
-	{{- else if .MapValue }}map[string]{{template "type" .MapValue}}
-	{{- else if .IsObject }}{{.PascalName}}
-	{{- else if .Enum }}types.String
-	{{- else}}any /* MISSING TYPE */
-	{{- end -}}
+    {{- if not . }}any /* ERROR */
+    {{- else if .IsExternal }}{{.Package.Name}}.{{.PascalName}}
+    {{- else if .IsAny}}any
+    {{- else if .IsEmpty}}[]{{.PascalName}}
+    {{- else if .IsString}}types.String
+    {{- else if .IsBool}}types.Bool
+    {{- else if .IsInt64}}types.Int64
+    {{- else if .IsFloat64}}types.Float64
+    {{- else if .IsInt}}types.Int64
+    {{- else if .IsByteStream}}io.ReadCloser
+    {{- else if .ArrayValue }}
+        {{- if .ArrayValue.IsObject }}{{template "type" .ArrayValue}}
+        {{- else }}[]{{template "type" .ArrayValue}}
+        {{- end }}
+    {{- else if .MapValue }}map[string]{{template "type" .MapValue}}
+    {{- else if .IsObject }}[]{{.PascalName}}
```

Comment:
Let's see if we can somehow capture some metadata in struct tag or some other way to capture max_items = 1.

Reply:
Yes, trying to figure out how to do this, will talk to Hector about it.